tests/callbacks/test_rich_progress_bar.py

```diff
@@ -44,6 +44,25 @@ def test_rich_progress_bar_refresh_rate():
     assert progress_bar.is_disabled
 
 
+@RunIf(rich=True)
+def test_rich_progress_bar_refresh_rate(tmpdir):
+    """Test that the refresh rate is set correctly based on the Trainer, and warn if the user sets the argument."""
+    trainer = Trainer(default_root_dir=tmpdir)
+    assert trainer.progress_bar_callback.refresh_rate_per_second == 10
+    assert isinstance(trainer.progress_bar_callback, RichProgressBar)
+
+    trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=None)
+    assert isinstance(trainer.progress_bar_callback, RichProgressBar)
+    assert trainer.progress_bar_callback.refresh_rate_per_second == 10
+
+    with pytest.warns(
+        UserWarning, match="``RichProgressBar`` does not support setting the refresh rate via the Trainer. "
+    ):
+        trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=19)
+        assert isinstance(trainer.progress_bar_callback, ProgressBar)
+        assert trainer.progress_bar_callback.refresh_rate == 19
```

Comment:
Let's put under `with` only the line which supposes to raise warning not all... this makes it a bit foggy